crates/proof-of-sql/src/proof_primitive/hyrax/base/hyrax_helpers.rs

```diff
@@ -0,0 +1,187 @@
+            let (hi_eighth6, hi_eighth7) = hi_quarter3.split_at_mut(1 << (nu - 3));
+            // Fill up quarter #2 (from 2/4..3/4).
+            hi_eighth3
+                .iter()
+                .zip(hi_eighth4.iter_mut().zip(hi_eighth5))
+                .for_each(|(&source, (target_lo, target_hi))| {
+                    // Copy eighth #3 (from 3/8..4/8) to eighth #4 (4/8..5/8).
+                    *target_lo = source;
+                    // Copy eighth #3 (from 3/8..4/8) to eighth #5 (5/8..6/8)
+                    // and multiply by the third from the last element in point.
+                    *target_hi = source * point[2 * nu - 4];
+                });
+            if add_last_quarter {
+                // Fill up quarter #4 (from 3/4..4/4).
+                hi_quarter2
+                    .iter()
+                    .step_by(2)
+                    .zip(hi_eighth6.iter_mut().zip(hi_eighth7))
+                    .for_each(|(&source, (target_lo, target_hi))| {
+                        // Copy every other in quarter #2 (from 2/4..3/4) to eighth #6 (6/8..7/8).
+                        *target_lo = source;
+                        // Copy every other in quarter #2 (from 2/4..3/4) to eighth #6 (7/8..8/8).
+                        // and multiply by the second from the last element in point.
+                        *target_hi = source * point[2 * nu - 3];
+                    });
+            }
+        }
+    }
+}
+
+#[allow(dead_code)]
+pub(super) fn fold_dynamic_standard_basis_tensors<S: Scalar>(
+    point: &[S],
+    alphas: &[S],
+    alpha_invs: &[S],
+) -> (S, S) {
+    let nu = point.len() / 2 + 1;
+    debug_assert_eq!(alphas.len(), nu);
+    debug_assert_eq!(alpha_invs.len(), nu);
+    let lo_fold = if point.is_empty() {
+        alphas[0]
+    } else {
+        point.iter().zip(alphas).map(|(v, a)| *v + *a).product()
+    };
+    let hi_fold = point
+        .iter()
+        .enumerate()
+        .fold(
+            (alpha_invs[0] + S::ONE, S::ZERO),
+            |(acc, prev_partial), (i, &p)| {
+                if i == 0 {
+                    (acc, S::ZERO)
+                } else if i == 1 {
+                    (acc * alpha_invs[1] + p * alpha_invs[0], S::ZERO)
+                } else if i % 2 == 0 {
+                    let partial = (i / 2 + 1..i)
+                        .zip(alpha_invs)
+                        .map(|(k, a)| point[k] + *a)
+                        .product();
+                    (acc + p * partial, partial)
+                } else {
+                    (
+                        acc * alpha_invs[i / 2 + 1]
+                            + p * alpha_invs[i / 2]
+                                * (point[i - 1] + alpha_invs[i / 2 - 1])
+                                * prev_partial,
+                        S::ZERO,
+                    )
+                }
+            },
+        )
+        .0;
+    (lo_fold, hi_fold)
+}
```

Comment:
Why is this dead? If its needed for something later on, please document it as such. If its no longer needed and wont be used, lets remove it for now.